File: src/main/java/com/example/ajaxonboardingproject/service/mapper/UserMapper.java
package com.example.ajaxonboardingproject.service.mapper;

import com.example.ajaxonboardingproject.dto.response.UserResponseDto;
import com.example.ajaxonboardingproject.model.User;
import org.springframework.stereotype.Component;

@Component
public class UserMapper implements ResponseDtoMapper<UserResponseDto, User> {
    @Override
    public UserResponseDto mapToDto(User user) {
        return new UserResponseDto(
                user.getId(),
                user.getEmail());
    }
}

File: src/main/java/com/example/ajaxonboardingproject/service/mapper/UserMapper.kt
package com.example.ajaxonboardingproject.service.mapper

import com.example.ajaxonboardingproject.dto.response.UserResponseDto
import com.example.ajaxonboardingproject.model.User
import org.springframework.stereotype.Component

@Component
class UserMapper : ResponseDtoMapper<UserResponseDto, User> {
    override fun mapToDto(model: User): UserResponseDto {
        return UserResponseDto(
            id = model.id,
            email = model.email
        )
    }
}
